ConsoleApp/Stack.cs

```diff
@@ -0,0 +1,69 @@
+namespace ConsoleApp;
+class LinkedStack<T>
+{
+    private record StackItem
+    {
+        public T Value { get; init; }
+        public StackItem Next { get; set; }
+    }
+
+    private StackItem _head;
+    public int Length { get; private set; }
+
+
+    public void Push(T value)
+    {
+        if (_head == null)
+        {
+            throw new NullReferenceException();
+        }
+        var newItem = new StackItem
+        {
+            Value = value
+        };
+        if (_head == null)
```

Comment:
if you remove this branch and leave only what goes after if - it will works for all scenarios